File: backend/Services/Parsers/RedactleParser.cs
using System.Text.RegularExpressions;
using backend.Models;

namespace backend.Services.Parsers
{
    public class RedactleParser : ResultParser
    {
        private ILogger<RedactleParser> _logger;

        public RedactleParser(ILogger<RedactleParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => false;
        public override string GameName => "Redactle";
        public override bool GolfScoring => true;
        public override string? HelpText => null;
        private const string ScoreGroup = "score";
        protected override Regex Parser => new Regex($@"^I solved {GameName} #\d+ in (?<{ScoreGroup}>\d+) guesses");
        public override string Url => "https://redactle.net/";

        protected override string GetCleanResult(string result, Match parserResults)
        {
            return result.Replace($"Play at {Url}", string.Empty).Trim();
        }

        public override string? GetScoreValue(DailyResult dailyResult)
        {
            return dailyResult.Score?.ToString();
        }

        protected override DailyResult SetScore(DailyResult dailyResult, Match parserResults)
        {
            if (Int32.TryParse(parserResults.Groups[ScoreGroup].Value, out var score))
            {
                dailyResult.Score = score;
            }

            return dailyResult;
        }
    }
}

File: lambdas/src/parsers/redactle-parser.ts
import { DailyResult } from '../models/daily-result';
import { ResultParser, getGroup, hasGroup } from './base/result-parser';

export class RedactleParser extends ResultParser {
  readonly countWinner = true;
  readonly gameName = 'Redactle';
  readonly golfScoring = true;
  readonly helpText = null;
  readonly url = 'https://redactle.net/';
  
  private static readonly SCORE_GROUP = 'score';
  
  // Match "I solved Redactle #769 in 38 guesses"
  protected readonly parser = new RegExp(
    `^I solved ${this.gameName} #\\d+ in (?<${RedactleParser.SCORE_GROUP}>\\d+) guesses`
  );
  
  protected getCleanResult(result: string, match: RegExpMatchArray): string {
    return result.replace(`Play at ${this.url}`, '').trim();
  }
  
  getScoreValue(dailyResult: DailyResult): string | null {
    return dailyResult.score?.toString() ?? null;
  }
  
  protected setScore(dailyResult: DailyResult, match: RegExpMatchArray): DailyResult {
    if (hasGroup(match, RedactleParser.SCORE_GROUP)) {
      const scoreStr = getGroup(match, RedactleParser.SCORE_GROUP);
      if (scoreStr) {
        const score = parseInt(scoreStr, 10);
        if (!isNaN(score)) {
          dailyResult.score = score;
        }
      }
    }
    return dailyResult;
  }
}
